### PR TITLE
0xarthurxyz/add mermaid diagram plugin

### PR DESCRIPTION
**Pull request by 0xarthurxyz**
### TLDR

- This small PR adds support for [Mermaid diagrams](https://mermaid-js.github.io/mermaid/#/) in `.md` files across [docs.celo.org](https://docs.celo.org/). 
- Checks: 
  - docs build as expected ✅
  - Mermaid diagrams render as expected ✅ (tested in these 2 commits: [1](https://github.com/celo-org/docs/commit/f3856fff4194447dc530c16f04a846cef2bf8619) and [2](https://github.com/celo-org/docs/commit/c9770a284f4b999b691b63173540d42dd55d5c71))

### Context 

- I would like to include diagrams and visualisations in the Celo docs to make understanding the Celo stack easier.
  - For example, we regularly refer to the [ODIS request flow diagram](https://docs.celo.org/celo-codebase/protocol/odis#request-flow-diagram), to explain fairly complicated flows more simply (see screenshot)
  - image
- I like to use [mermaid diagrams](https://mermaid-js.github.io/mermaid/#/) for that (they are natively supported and render in [Github](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/), [Notion](https://www.redgregory.com/notion/2022/1/17/how-to-get-started-with-mermaid-in-notion) and possibly [out-of-the-box in docusaurus](https://github.com/facebook/docusaurus/issues/1258) soon)
  - To use them, you can simple include the _instructions_ for the diagram as “code” encapsulated in back-ticks \```mermaid [...] ``` and it renders automatically (see screenshot)
  - image

### Why is this useful?

- Because it saves contributors to the docs time creating diagrams elsewhere, exporting a `.png` version of them, importing them in markdown files and managing various `.png` files in an `/assets` folder somewhere
- Instead, contributors can simply include the diagram in back-ticks, make a small modification within an `.md` file (when needed), push the code and the diagrams render immediately.

### How?

- There is a [mermaid plugin](https://github.com/sjwall/mdx-mermaid) for docusaurus [listed in the official docusaurus docs](https://docusaurus.io/community/resources)
- I followed the instructions in their docs [here](https://sjwall.github.io/mdx-mermaid/docs/intro/) to modify
  - `packages.json`, and
  - `docusaurus.config.js`.

### CODE REVIEW

**Review by joenyzio — APPROVED** [review]

**Comment by 0xarthurxyz** [comment]
Thanks for the quick review @joenyzio 🙏
